File: app/src/main/java/de/squiray/dailytodo/domain/usecase/NoOpResultHandler.java
package de.squiray.dailytodo.domain.usecase;

public abstract class NoOpResultHandler<T> implements ResultHandler<T> {
	@Override
	public void onFinished() {
		// no-op
	}

	@Override
	public void onError(Throwable e) {
		// no-op
	}

	@Override
	public void onSuccess(T t) {
		// no-op
	}
}

File: app/src/main/java/de/squiray/dailytodo/domain/usecase/NoOpResultHandler.kt
package de.squiray.dailytodo.domain.usecase

abstract class NoOpResultHandler<T> : ResultHandler<T> {
    override fun onFinished() {
        // no-op
    }

    override fun onError(e: Throwable) {
        // no-op
    }

    override fun onSuccess(t: T) {
        // no-op
    }
}
